src/libs/actions/Task.ts

```diff
@@ -312,9 +320,7 @@ function createTaskAndNavigate(
 
     API.write(WRITE_COMMANDS.CREATE_TASK, parameters, {optimisticData, successData, failureData});
 
-    ReportConnection.updateReportData(parentReportID, {
-        lastReadTime: currentTime,
-    });
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${parentReportID}`, {lastReadTime: currentTime});
```

Comment:
This change is different from the previous behaviour in ReportConnection.ts, [where we update the local `allReports` instead of doing Onyx operations](https://github.com/Expensify/App/blob/main/src/libs/ReportConnection.ts#L58-L69). It would be good to understand why we were doing that before and if it's safe to use Onyx operations instead.

Reply:
It looks like it will be removed anyway: https://github.com/Expensify/App/pull/51174/files#r1866068419

Reply:
Nevertheless, I've [asked the author](https://github.com/Expensify/App/pull/51174/files#r1869869695) why this approach was selected initially.

Reply:
Overall the `ReportConnection.updateReportData` function, as well as some other changes including [reportLastReadTime](https://github.com/Expensify/App/pull/51174/files#diff-42d45a343b031265784f22ea27e4502b3c1968e492b1fc83154414b584d59217R214) memoized value were introduced in this [PR](https://github.com/Expensify/App/pull/51174/files) to fix this issue: https://github.com/Expensify/App/issues/50792.
I've tested the issue over my PR and it works as expected. 
I've just tried to remove the updates raised in that PR as it was [recommended](https://github.com/Expensify/App/pull/51174/files#r1870917774), and there is no issue for me as well. So I think we can get rid of it, did it in: https://github.com/Expensify/App/pull/53385/commits/d01ac8d7677dd9bd7305dd8215fb99f3aa5e62b3